src/app/space/analyze/analyze-overview/analyze-overview.spec.ts

```diff
@@ -0,0 +1,228 @@
+import { AuthenticationService, User, UserService } from 'ngx-login-client';
+import { ConnectableObservable, Subscription } from 'rxjs';
+import { Observable, Subject } from 'rxjs';
+import { createMock } from 'testing/mock';
+import {
+  initContext,
+  TestContext
+} from 'testing/test-context';
+
+@Component({
+  template: '<alm-analyzeOverview></alm-analyzeOverview>'
+})
+class HostComponent { }
+
+@Component({
+  selector: 'fabric8-edit-space-description-widget',
+  template: ''
+})
+class FakeFabric8EditSpaceDescriptionWidget {
+}
+
+@Component({
+  selector: 'fabric8-work-item-widget',
+  template: ''
+})
+class FakeFabric8WorkItemWidget {
+}
+
+@Component({
+  selector: 'fabric8-pipelines-widget',
+  template: ''
+})
+class FakeFabric8PipelinesWidget {
+}
+
+@Component({
+  selector: 'fabric8-add-codebase-widget',
+  template: ''
+})
+class FakeFabric8AddCodebaseWidget {
+}
+
+@Component({
+  selector: 'fabric8-analytical-report-widget',
+  template: ''
+})
+class FakeFabric8AnalyticalReportWidget {
+}
+
+@Component({
+  selector: 'fabric8-create-work-item-widget',
+  template: ''
+})
+class FakeFabric8CreateWorkItemWidget {
+}
+
+@Component({
+  selector: 'fabric8-environment-widget',
+  template: ''
+})
+class FakeFabric8EnvironmentWidget {
+}
+
+@Component({
+  selector: 'f8-feature-toggle',
+  template: ''
+})
+class FakeFabric8FeatureToggle {
+}
+
+@Component({
+  selector: 'flow-selector',
+  template: ''
+})
+class FlowSelector {
+  @Input() space: Space;
+}
+
+@Component({
+  selector: 'import-wizard',
+  template: ''
+})
+class ImportWizard {
+}
+
+@Component({
+  selector: 'quickstart-wizard',
+  template: ''
+})
+class QuickstartWizard {
+}
+
+describe('AnalyzeOverviewComponent', () => {
+  type ctx = TestContext<AnalyzeOverviewComponent, HostComponent>;
+
+  let modalService: jasmine.SpyObj<BsModalService>;
+  let broadcaster: jasmine.SpyObj<Broadcaster>;
+  let authentication: jasmine.SpyObj<AuthenticationService>;
+  let ctxSubj: Subject<Context>;
+
+  let authenticationServiceMock = (): jasmine.SpyObj<AuthenticationService> => {
+    authentication.isLoggedIn = function() { return true; } as (() => boolean) & jasmine.Spy;
+    return authentication;
+  };
+
+  let contextsMock = (): jasmine.SpyObj<Contexts> => {
+    const contexts: jasmine.SpyObj<Contexts> = createMock(Contexts);
+
+    ctxSubj = new Subject<Context>();
+    contexts.current = ctxSubj as Subject<Context> & jasmine.Spy;
+
+    return contexts;
+  };
+
+  let userServiceMock = () => {
+    const userService: jasmine.SpyObj<UserService> = createMock(UserService);
+    userService.loggedInUser = Observable.of({
+      id: 'loggedInUser'
+    } as User).publish() as ConnectableObservable<User> & jasmine.Spy;
+
+    return userService;
+  };
+
+  beforeEach(() => {
+    modalService = createMock(BsModalService);
+    broadcaster = createMock(Broadcaster);
+    authentication = createMock(AuthenticationService);
```

Comment:
Using a beforeEach here to create the mocks and then using a useFactory below to return a reference to the freshly created instance is awkward. Why not just have the factory call createMock itself and return a new instance directly? Or, since you don't seem to be making any assertions about any of the functions on these mocks, why not just use `useValue` and return one instance once without using a factory at all?